components/Proposals/MintConditionalTokenCard.tsx

```diff
@@ -1,24 +1,43 @@
-import { useCallback, useState } from 'react';
-import { Button, Fieldset, Group, Text, TextInput } from '@mantine/core';
+import { useCallback, useEffect, useState } from 'react';
+import { Button, Fieldset, Group, Text, TextInput, SegmentedControl, Center, Box } from '@mantine/core';
 import numeral from 'numeral';
 import Link from 'next/link';
-import { IconExternalLink } from '@tabler/icons-react';
-import { Token } from '@/hooks/useTokens';
+import { IconExternalLink, IconCurrencyDollar, IconLetterM } from '@tabler/icons-react';
+// import { Token } from '@/hooks/useTokens';
 import { useProposal } from '@/contexts/ProposalContext';
 import { useTransactionSender } from '../../hooks/useTransactionSender';
 import { useExplorerConfiguration } from '@/hooks/useExplorerConfiguration';
 import { useBalance } from '../../hooks/useBalance';
 import { NUMERAL_FORMAT } from '../../lib/constants';
+import { useTokens, Token } from '@/hooks/useTokens';
 
-export function MintConditionalTokenCard({ token }: { token: Token }) {
+export function MintConditionalTokenCard() {
   const sender = useTransactionSender();
   const { markets, mintTokensTransactions } = useProposal();
   const { generateExplorerLink } = useExplorerConfiguration();
+  const { tokens } = useTokens();
 
   if (!markets) return null;
 
   const [mintAmount, setMintAmount] = useState<number>();
-  const fromBase = markets.baseVault.underlyingTokenMint.equals(token.publicKey);
+  const [token, setToken] = useState<string>('meta');
```

Comment:
we could use selectedTokenName and token, might be a bit more clear

Reply:
Also, what's the point of using two variables instead of only storing a token and using its name attributes when needed?

Reply:
Totally, I think it was because I couldn't get typing to work and I revert back to "how can I get this to work" vs "this is the right way to do it" but yep will circle back on this.